llvm/lib/SandboxIR/SandboxIR.cpp

```diff
@@ -668,6 +710,11 @@ Value *Context::getOrCreateValueInternal(llvm::Value *LLVMV, llvm::User *U) {
     It->second = std::unique_ptr<StoreInst>(new StoreInst(LLVMSt, *this));
     return It->second.get();
   }
+  case llvm::Instruction::Ret: {
+    auto *LLVMRet = cast<llvm::ReturnInst>(LLVMV);
+    It->second = std::unique_ptr<ReturnInst>(new ReturnInst(LLVMRet, *this));
+    return It->second.get();
+  }
   default:
```

Comment:
Can you replace with:

auto lambda = []<typename T>(auto It, auto LLVMV) {
  auto *LLVMVal = cast<T>(LLVMV);
  It->second = std::unique_ptr<T>(new T(...));
  return It->second.get();
};

case llvm::Instruction::Ret: {
           return lambda<ReturnInst>(It);
}

Reply:
I meant creating a polymorphic lambda here, that didn't come out like I had typed.

Reply:
That would be really nice, but I am not sure it can be done without a template. Because we need a cast: `auto *LLVMI = cast<LLVMT>(LLVM)` and a new: `new SBT(LLVMI, *this)`, where both `LLVMT` and `SBT` are types. I don't think I can get these types some other way using decltype.